src/tasks/MonoTargetsTasks/ILStrip/ILStrip.cs

```diff
@@ -81,4 +114,192 @@ private bool StripAssembly(ITaskItem assemblyItem)
+            {
+                Log.LogError($"{assemblyFilePath} read from {methodTokenFile} doesn't exist.");
+                return true;
+            }
+
+            string trimmedAssemblyFilePath = ComputeTrimmedAssemblyPath(assemblyFilePath);
+            bool isTrimmed = false;
+
+            using FileStream fs = File.Open(assemblyFilePath, FileMode.Open);
+            {
+                using PEReader peReader = new(fs, PEStreamOptions.LeaveOpen);
+                MetadataReader mr = peReader.GetMetadataReader();
+
+                string guidValue = ComputeGuid(mr);
+                string? expectedGuidValue = sr.ReadLine();
+                if (!string.Equals(guidValue, expectedGuidValue, StringComparison.OrdinalIgnoreCase))
+                {
+                    Log.LogError($"[ILStrip] GUID value of {assemblyFilePath} doesn't match the value listed in {methodTokenFile}.");
+                    return true;
+                }
+
+                string? line = sr.ReadLine();
+                if (!string.IsNullOrEmpty(line))
+                {
+                    isTrimmed = true;
+                    Dictionary<int, int> method_body_uses = ComputeMethodBodyUsage(mr, sr, line, methodTokenFile);
+                    CreateTrimmedAssembly(peReader, trimmedAssemblyFilePath, fs, method_body_uses);
+                }
+            }
+            if (isTrimmed)
+            {
+                AddItemToTrimmedList(assemblyFilePath, trimmedAssemblyFilePath);
+            }
+        }
+
+        return true;
+    }
+
+    private static string ComputeTrimmedAssemblyPath(string assemblyFilePath)
+    {
+        string? assemblyPath = Path.GetDirectoryName(assemblyFilePath);
+        string? assemblyName = Path.GetFileNameWithoutExtension(assemblyFilePath);
+        if (string.IsNullOrEmpty(assemblyPath))
+        {
+            return (assemblyName + "_new.dll");
+        }
+        else
+        {
+            return Path.Combine(assemblyPath, (assemblyName + "_new.dll"));
+        }
+    }
+
+    private static string ComputeGuid(MetadataReader mr)
+    {
+        GuidHandle mvidHandle = mr.GetModuleDefinition().Mvid;
+        Guid mvid = mr.GetGuid(mvidHandle);
+        return mvid.ToString();
+    }
+
+    private Dictionary<int, int> ComputeMethodBodyUsage(MetadataReader mr, StreamReader sr, string? line, string methodTokenFile)
+    {
+        Dictionary<int, int> token_to_rva = new();
+        Dictionary<int, int> method_body_uses = new();
+
+        foreach (MethodDefinitionHandle mdefh in mr.MethodDefinitions)
+        {
+            int methodToken = MetadataTokens.GetToken(mr, mdefh);
+            MethodDefinition mdef = mr.GetMethodDefinition(mdefh);
+            int rva = mdef.RelativeVirtualAddress;
+
+            token_to_rva.Add(methodToken, rva);
+
+            if (method_body_uses.TryGetValue(rva, out var _))
+            {
+                method_body_uses[rva]++;
+            }
```

Comment:
When will this be true? Wouldn't all the `methodDefs` have unique RVAs?

Reply:
No there are cases that methods share the same rva.